paasta_tools/kubernetes_tools.py

```diff
@@ -3432,3 +3434,44 @@ def update_crds(
             success = False
 
     return success
+
+
+def get_kubernetes_secret_name(service_name, secret_name, namespace="paasta"):
+    service = sanitise_kubernetes_name(service_name)
+    sanitised_secret = sanitise_kubernetes_name(secret_name)
+    name = f"{namespace}-secret-{service}-{sanitised_secret}"
+    return name
+
+
+def get_kubernetes_secret(secret_name, service_name, cluster):
```

Comment:
I'd make this method just accept a single cluster as a string, rather than expecting a list and then lookup index-0.